### PR TITLE
bomctl - OpenSSF Sandbox Application

### PR DESCRIPTION
**Pull request by idunbarh**
The __bomctl__ project was developed as an experiment within the Security Tooling WG as a response to a [2023 Secure Open Source Software Summit item](https://github.com/ossf/wg-security-tooling/issues/61). We are seeking admission to the OpenSSF sandbox and continued alignment to the Security Tooling WG.

## Overview

__bomctl__ is format-agnostic Software Bill of Materials (SBOM) tooling, which is intended to bridge the gap between SBOM generation and SBOM analysis tools. It focuses on supporting more complex SBOM operations by being opinionated on only supporting the [NTIA minimum fields](https://www.ntia.doc.gov/files/ntia/publications/sbom_minimum_elements_report.pdf) or other fields supported by [protobom](https://github.com/protobom/protobom).

It is intended to help developers who need to manipulate SBOMs at the CLI or within a workflow. Example operations would be merging in project specific SBOM data that would not be detected by a SBOM generation tool.

### List of project maintainers

The project must have a minimum of three maintainers with a minimum of two different organizational affiliations.

* "Jonathan Howard", "Lockheed Martin", @jhoward-lm
* "Eddie Zaneski", "Defense Unicorns", @eddiezane
* "Allen Shearin", "Lockheed Martin", @ashearin
* "Ian Dunbar-Hall", "Lockheed Martin", @idunbarh

### Sponsor

Most projects will report to an existing OpenSSF Working Group, although in some cases a project may report directly to the TAC. The project commits to providing quarterly updates on progress to the group they report to.

* Security Tooling WG - This project was initial developed as an experiment under the Security Tooling WG.

### Mission of the project

The project must be aligned with the OpenSSF mission and either be a novel approach for existing areas, address an unfulfilled need, or be initial code needed for OpenSSF WG work. It is preferred that extensions of existing OpenSSF projects collaborate with the existing project rather than seek a new project.

* "__bomctl__ is format-agnostic Software Bill of Materials (SBOM) tooling, which is intended to bridge the gap between SBOM generation and SBOM analysis tools. It focuses on supporting more complex SBOM operations by being opinionated on only supporting the NTIA minimum fields or other fields supported by protobom."
* "__bomctl__ started as an action from Secure Open Source Summit 2023 in DC. The action item was to merge existing sbom tooling into a single source for sbom format agnostic tooling to manage relationships between SBOM files. No existing tooling beyond the [protobom](https://github.com/protobom/protobom) work existed that was format agnostic. This lead to __bomctl__ being developed by the Security Tooling WG."
* "__bomctl__ heavily builds on [protobom](https://github.com/protobom/protobom)."
  * "[Protobom](https://github.com/protobom/protobom) is a go library for manipulating SBOM data in an agnostic manner"
  * "__Bomctl__ is a CLI that handles movement, operations, and caching of SBOM files and will allow linkages between SBOM files"

#### Specific Goals Include:

* Simplify the process of manipulate SBOM Files while being SBOM format agnostic
* Simplify linking SBOM Files to allow "trees" of SBOM Files to handle capturing systems of systems
* Manage reading SBOM Files from a variety of sources
* Manage writing SBOM Files to a variety of destinations
* Create a CLI to wrap protobom go library functionality 

#### Non-Goals Include:

* Enriching SBOM from data sources
* Providing Software Supply Chain Insights (please use GUAC for this)
* Managing SBOMs "at scale"

### IP policy and licensing due diligence

When contributing an existing Project to the OpenSSF, the contribution must undergo license and IP due diligence by the Linux Foundation (LF).

* "TBD" - https://github.com/ossf/tac/issues/368
  
### Project References

The project should provide a list of existing resources with links to the repository, and if available, website, a roadmap, contributing guide, demos and walkthroughs, and any other material to showcase the existing breadth, maturity, and direction of the project.

| Reference           | URL |
|---------------------|-----|
| Repo                | https://github.com/bomctl/bomctl |
| Website             | https://github.com/bomctl/bomctl |
| Contributing guide  | https://github.com/bomctl/bomctl/blob/main/CONTRIBUTING.md |
| Security.md         | https://github.com/bomctl/bomctl/blob/main/SECURITY.md |
| Scorecard Report    | https://securityscorecards.dev/viewer/?uri=github.com/bomctl/bomctl |

### CODE REVIEW

File: process/project-lifecycle-documents/bomctl_sandbox_stage.md
Comment (marcelamelara) [inline process/project-lifecycle-documents/bomctl_sandbox_stage.md:25]: Nit
```suggestion
  * "[Protobom](https://github.com/protobom/protobom) is a go library for manipulating SBOM data in a schema-agnostic manner"
```

**Review by marcelamelara — COMMENTED** [review]
Thanks @idunbarh ! I support admitting bomctl as a new sandbox project! I'll approve once the IP policy and licensing due diligence is completed.

**Comment by SecurityCRob** [comment]
I also agree with this and will approve once legal review is complete.

**Comment by hythloda** [comment]
The legal review has significant backlog and might take a few weeks to complete. Because the announcement is desired at the beginning of September can we shift this to getting a vote from all TAC members for the following options:

- [ ] Approve pending no concerns from license review
- [ ] Waiting until license review for my vote
- [ ] Needs improvements (comments)
- [ ] Denied

**Comment by mlieberman85** [comment]
Approved pending no concerns from license review

**Comment by camaleon2016** [comment]
Approve pending no concerns from license review

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Review by steiza — APPROVED** [review]
Approve pending no concerns from license review

**Comment by bobcallaway** [comment]
Approve pending no concerns from license review

**Review by marcelamelara — COMMENTED** [review]
Approve pending no concerns from license review

**Comment by SecurityCRob** [comment]
Approve pending no concerns from license review

**Comment by torgo** [comment]
Approve pending no concerns from license review

**Review by sevansdell — APPROVED** [review]
Approve pending no issues in legal review.

**Review by lehors — APPROVED** [review]
Approve pending no concerns from license review

**Review by SecurityCRob — APPROVED** [review]
i approve pending any outstanding ip/legal review feedback

**Comment by hythloda** [comment]
The IP and license review concluded with no issues. 
We can close this out as approved 🎉

**Review by marcelamelara — APPROVED** [review]
Since the license and IP review came back with no issues, I approve admitting bomctl as an OpenSSF sandbox project!

**Review by mlieberman85 — APPROVED** [review]
Approved given no license issues.

**Comment by david-a-wheeler** [comment]
The majoriy of the TAC has approved, so this is complete! Congrats & welcome to OpenSSF!

**Comment by david-a-wheeler** [comment]
@SecurityCRob - as TAC chair, do you want to officially push the merge button?

**Comment by lehors** [comment]
TAC approval on call recorded in the [2024-08-20 TAC minutes](https://docs.google.com/document/d/1-zrtagRnPd75TDT1zRxrtxE9SpMIBJdPmaolaw4woQA/edit#heading=h.8t68mnoyuq7x)

**Pull request merged**
